base/collect_grpc.go

```diff
@@ -64,25 +72,72 @@ func (t *collectGRPCTask) validateInputs() error {
 }
 
 // resultForVersion collects gRPC test result for a given version
-func (t *collectGRPCTask) resultForVersion() (*runner.Report, error) {
+func (t *collectGRPCTask) resultForVersion() (map[string]*runner.Report, error) {
 	// the main idea is to run ghz with proper options
 
-	opts := runner.WithConfig(&t.With.Config)
+	var err error
+	results := map[string]*runner.Report{}
+
+	if len(t.With.Endpoints) > 0 {
```

Comment:
would it make sense to "create an endpoint" if there is none. Then processing is the same; no need for conditional run code.

Reply:
@kalantar Are you suggesting that during preprocessing, we will check if there is `Endpoints`, if not and if there is base configuration, then we will add that as an element to `Endpoints`. Then during the `run()` portion, we will not need this check and we can just iterate over `Endpoints`?